LayoutTests/fast/scrolling/mac/smooth-scroll-recursive-frame-to-frame.html

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ useFlexibleViewport=true EventHandlerDrivenSmoothKeyboardScrollingEnabled=true ] -->
+
+<html>
+
+<head>
+    <script src="../../../resources/ui-helper.js"></script>
+    <script src="../../../resources/js-test-pre.js"></script>
+    <meta name="viewport" content="initial-scale=1.5, user-scalable=no">
+    <style>
+        body {
+            height: 2000px;
+        }
+        #iframe {
+            width: 20%;
+            height: 20%;
+            border: 1px solid black;
+            padding: 10px;
+        }
+    </style>
+    <script>
+        if (window.testRunner) {
+            testRunner.dumpAsText();
+            testRunner.waitUntilDone();
+        }
+
+        function documentScrolled()
+        {
+            debug("SUCCESSFUL: frame to frame recursive scroll.");
```

Comment:
This would be more clear as "keyboard scroll propagated from iframe to main document" if that's accurate.